### PR TITLE
Add CI blog post

### PR DESCRIPTION
**Pull request by cevich**
Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
***Note:*** First pass transcoding from a google doc. Got all the words, but still need to copy all the links and do a bit of formatting.

**Comment by TomSweeneyRedHat** [comment]
Trying a comment this way.  For some reason can't comment in line?

**Comment by rhatdan** [comment]
Lets publish on Monday.

File: _posts/2019-03-12-CI3.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+* Travis
+  * With [Ubuntu Trusty only a few years old](https://lists.ubuntu.com/archives/ubuntu-announce/2014-April/000182.html),
+    we ran tests on a platform version nobody was using,
+    with bleeding edge-code jammed on top.  Some OS-X tests ran, and we think at least one person
+    looked at the results, some of the time, every once in a while.
+  * Required a contrived containerized-environment to workaround host-side limitations.  Fixes for
+    fake environments almost never improve reality.  e.g. impossible to test or fix AppArmor or
+    SELinux problems from inside a container.
+  * Also not real world tests.  Most people would never run container tools within a container, and
```
Comment (TomSweeneyRedHat) [inline _posts/2019-03-12-CI3.md]: Weird, now GitHub is letting me comment in line.  "Also not real world tests." is kind of incomplete.  Perhaps:  "The tests run in Travis were not real world tests."

File: _posts/2019-03-12-CI3.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    certain security tools like SELinux and AppArmor would not be tested running inside of a
+    container.
+
+* PAPR
+  * An internal "maintenance mode" service, meaning only bug-fixes, no new features. Supported by a
+    single, talented engineer, from another group, perfectly happy to be working on something else.
```
Comment (TomSweeneyRedHat) [inline _posts/2019-03-12-CI3.md]: i'd soften "perfectly happy to be working on something else".  Maybe "but who had other higher priority commitments." or some such

File: _posts/2019-03-12-CI3.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+So
+[Cirrus CI gives all the power](https://cirrus-ci.org/#comparison-with-popular-ciaas)
+for success, and/or blasting giant, perfectly round, holes in your own two feet!
+Our CI experience can be as simple or complex as we like, and reliability will match that of major
+cloud providers and the inverse of our cleverness. What could possibly go wrong? :D
+
+### VM Image Orchestration
+
+Starting from the bowels of any stack quickly arrives at needing to orchestrate initial operating
```
Comment (TomSweeneyRedHat) [inline _posts/2019-03-12-CI3.md]: I'm having a hard time parsing this sentence.  who/what quickly arrives?  Perhaps "stack, one quickly arrives at the need"

File: _posts/2019-03-12-CI3.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+### Conclusion
+
+There are many other details I could get into, but sadly, my coffee mug is empty and I can see that I
+forgot to wash it (again).  Nevertheless, if you need some simple nuts-and-bolts automation, I
+highly recommend [Cirrus-CI](https://cirrus-ci.org).   It's (beer) free to use for open-source
+projects.  The Google Cloud Engine is also pseudo-free for quite a while, since they give you a
+very generous, and substantial startup credit.
+
+Other than finding a new mug or my soap, if there are any important specifics here,
```
Comment (TomSweeneyRedHat) [inline _posts/2019-03-12-CI3.md]: We don't have a comment section.  Suggest having them send you an e-mail or perhaps post an RFE issue to podman?

File: _posts/2019-03-12-CI3.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+        - $SCRIPT_BASE/integration_test.sh    # and this
+```
+
+With [Cirrus CI "installed"](https://cirrus-ci.org/guide/quick-start/)
+on a GitHub repository, upon any pull
+request change, Cirrus CI will step
+in to kick things within GCE, then report back results in your pull request.
+
+However, we also need to test more than one OS.  This is accomplished in Cirrus CI, by what they
+call a [matrix modifier](https://cirrus-ci.org/guide/writing-tasks/#matrix-modification).  That's
```
Comment (TomSweeneyRedHat) [inline _posts/2019-03-12-CI3.md]: it looks like they call it "matrix modification".  Perhaps "call [matrix modificiation]".

**Review by TomSweeneyRedHat — COMMENTED** [review]
Looks good overall, thanks for putting this together Chris.  A few suggestions here and there.  If you could please, change the date of the file to 3/18/2019 and after you update we'll push it out on Monday.

File: _posts/2019-03-12-CI3.md (hunk 2 above)
Comment (cevich) [reply _posts/2019-03-12-CI3.md]: This wording has blessing of the actual "talented engineer",  I can change it if you like, but prefer the more raw, "gritty" wording :grinning:

File: _posts/2019-03-12-CI3.md (hunk 5 above)
Comment (cevich) [reply _posts/2019-03-12-CI3.md]: I think this is a English-second-language problem in the docs (I'm assuming the author is non-native, and very busy with more important things).  I should probably post a PR to fix it...https://github.com/cirruslabs/cirrus-ci-docs/pull/204 though that's probably going to break my link :frowning_face:

File: _posts/2019-03-12-CI3.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+![podman logo](https://podman.io/images/podman.svg)
+
+{% assign author = site.authors[page.author] %}
+# CI, and CI, and CI, oh my! (then more CI)
+## By {{ author.display_name }} [GitHub](https://github.com/{{ author.github }})
+
+I wanted to write a detailed post about the CI setup we use for exercising proposed
+changes to [libpod (podman repo)](https://github.com/containers/libpod).  Unfortunately
+this topic (and automation in general)
+is so big, most  readers would end up on the floor, sound asleep, in a puddle of their
```
Comment (RDIL) [inline _posts/2019-03-12-CI3.md]:
```suggestion
is so big, most readers would end up on the floor, sound asleep, in a puddle of their
```
double space :o

**Comment by cevich** [comment]
@TomSweeneyRedHat lets wait a little and see where my Cirrus Docs PR goes (if anywhere), so we don't publish a 0-day broken link :grinning:

File: _posts/2019-03-12-CI3.md (hunk 2 above)
Comment (TomSweeneyRedHat) [reply _posts/2019-03-12-CI3.md]: Your call, I'm fine with it as is, any arrows coming won't be slung in my general direction!  ;^)

**Comment by TomSweeneyRedHat** [comment]
OK, give me a thumbs up when you want me to push the shiny button to publish, and I'll do so.  If that's Friday by chance, I'll wait until next Monday.

**Comment by cevich** [comment]
Mmkay, will do. Thx Tom.

**Comment by cevich** [comment]
@TomSweeneyRedHat okay, double-checking links, then I think this is good to go...

**Comment by cevich** [comment]
...okay, all the links work.  @TomSweeneyRedHat  okay by me to press the green button.

Thanks for all the help too  :smile:

**Comment by TomSweeneyRedHat** [comment]
LGTM
Thanks @cevich !

**Pull request merged**